addons/account/tests/test_transfer_wizard.py

```diff
@@ -289,3 +293,126 @@ def test_transfer_wizard_no_currency_conversion(self):
+        respect the bill algo for the lock period.
+        """
+
+        # Setup
+        self.company.expense_accrual_account_id = self.env['account.account'].create({
+            'name': 'Expense Accrual Account',
+            'code': '113226',
+            'user_type_id': self.env.ref('account.data_account_type_expenses').id,
+            'reconcile': True,
+        })
+        self.company.revenue_accrual_account_id = self.env['account.account'].create({
+            'name': 'Accrual Revenue Account',
+            'code': '226113',
+            'user_type_id': self.env.ref('account.data_account_type_revenue').id,
+            'reconcile': True,
+        })
+
+        misc_journal_id = self.company_data['default_journal_misc'].id
+        journal_id = self.env['account.journal'].create({
+            'name': 'Test Journal',
+            'code': 'TJ',
+            'type': 'general',
+        }).id
+
+        # move is in the lock period
+        move1 = self.env['account.move'].create({
+            'journal_id': journal_id,
+            'date': '2019-01-01',
+            'line_ids': [
+                Command.create({'account_id': self.accounts[0].id, 'debit': 1000, }),
+                Command.create({'account_id': self.accounts[0].id, 'credit': 1000, }),
+            ]
+        })
+        # move not in locked period, will do cut-off in the first month following lock period
+        move2 = self.env['account.move'].create({
+            'journal_id': journal_id,
+            'date': '2019-03-01',
+            'line_ids': [
+                Command.create({'account_id': self.accounts[0].id, 'debit': 2000, }),
+                Command.create({'account_id': self.accounts[0].id, 'credit': 2000, }),
+            ]
+        })
+        # move not in locked period, will do cut-off after the first month following lock period
+        move3 = self.env['account.move'].create({
+            'journal_id': journal_id,
+            'date': '2019-03-01',
+            'line_ids': [
+                Command.create({'account_id': self.accounts[0].id, 'debit': 3000, }),
+                Command.create({'account_id': self.accounts[0].id, 'credit': 3000, }),
+            ]
+        })
+        # testing with an invoice, because invoice hase a yearly reset sequence
+        invoice_1 = self.init_invoice(move_type='out_invoice', invoice_date='2019-01-02', amounts=[100])
+
+        (move1 + move2 + move3 + invoice_1).action_post()
+
+        move1.company_id.period_lock_date = fields.Date.from_string('2019-02-28')
+        move1.company_id.fiscalyear_lock_date = fields.Date.from_string('2019-02-28')
+
+        for move, date, wizard_date, expected_date in [
+            (move1, '2019-04-01', '2019-03-01', '2019-03-31'),
+            (move2, '2019-03-02', '2019-03-02', '2019-03-02'),
+            (move3, '2019-04-03', '2019-04-02', '2019-03-31'),
+            (invoice_1, '2019-06-02', '2019-06-01', '2019-03-31'),
```

Comment:
Since the move itself doesn't matter, only the date, why don't you put the date in the tuple and create it in the loop?